Выполнено:

BI003 opened the pull request:
а) наполнить базу тестовыми данными пользуясь оператором Insert; б) провести выборки клиентов, у которых сумма на счету ниже определенного значения, отсортированных в порядке возрастания суммы; в) провести поиск клиента с минимальной суммой на счете; г) провести подсчет суммы денег у всех клиентов банка; д) с помощью оператора Join, получить выборку банковских счетов и их владельцев;
е) вывести клиентов от самых старших к самым младшим; ж) подсчитать количество человек, у которых одинаковый возраст; з) сгруппировать клиентов банка по возрасту;
и) вывести только N человек из таблицы.

BI003 committed:
Выполнено:
ф) наполнить базу тестовыми данными пользуясь оператором Insert;
б) провести выборки клиентов, у которых сумма на счету ниже
определенного значения, отсортированных в порядке возрастания суммы;
в) провести поиск клиента с минимальной суммой на счете;
г) провести подсчет суммы денег у всех клиентов банка;
д) с помощью оператора Join, получить выборку банковских счетов и
их владельцев;
е) вывести клиентов от самых старших к самым младшим;
ж) подсчитать количество человек, у которых одинаковый возраст;
з) сгруппировать клиентов банка по возрасту;
и) вывести только N человек из таблицы.

BI003 committed:
Попытка отправки 2

PetrovskiAlex commented on Practice/console_1.sql:
```diff
@@ -0,0 +1,45 @@
+insert into client (id, name, surname, age, passport, phone_number, date_of_birth)
```
а где sql создания таблиц?

PetrovskiAlex commented on Practice/console_1.sql:
```diff
@@ -0,0 +1,45 @@
+
+INSERT INTO account (amount, "currency_id ", "client_id ")
+VALUES
+(1, '83fd2091-eeec-4df2-aa16-b6bfd202cad7', 'ba2c1141-a337-49c0-bbea-38fa349a071c'),
```
при попытке накататить этот скрипт буду ошибки из-за хардкода гуидов

PetrovskiAlex commented on Practice/console_1.sql:
```diff
@@ -0,0 +1,45 @@
+
+select name, surname, amount from  client
+join account a on client.id = a."client_id "
+order by amount limit 1;
```
а если таких клиентов несколько?

PetrovskiAlex commented on Practice/console_1.sql:
```diff
@@ -0,0 +1,45 @@
+
+select age, count(*) as same_age_people from client
+group by age
+having count(*)>1;
```
зачем having?

BI003 replied:
having нужен, чтобы приходили только те возраста, у которых больше одной записи.

PetrovskiAlex commented on Practice/console_1.sql:
```diff
@@ -0,0 +1,45 @@
+join currency c on a."currency_id " = c.id;
+
+select name, surname, age from client
+order by age DESC;
```
не в ту сторону сортировка

BI003 replied:
Почему? Задание: "е) вывести клиентов от самых старших к самым младшим"
![изображение_2024-10-13_005923948](https://github.com/user-attachments/assets/c08269a7-da6f-433d-b273-302c69b63084)

BI003 committed:
Добавил sql создания таблиц.
Изменил запрос на вывод клиентов, чья сумма на счёте равна минимальной среди всех записей в таблице account. Теперь если больше 1 клиента имеют одинаковую минимальную зарплату, то выведутся они оба.
Изменил запрос на заполнение account, теперь id генерируются автоматически.
